### PR TITLE
Add options to clear authorization headers for basic/bearer auth

### PR DESCRIPTION
**Pull request by liggitt**

### CODE REVIEW

**Comment by liggitt** [comment]
@deads2k

File: pkg/auth/authenticator/request/paramtoken/paramtoken.go [hunk 1]
```diff
@@ -13,16 +13,16 @@ import (
...
-	// Param is the query param to use as a token
+	// param is the query param to use as a token
 	param string
-	// Auth is the token authenticator to use to validate the token
+	// auth is the token authenticator to use to validate the token
 	auth authenticator.Token
-	// Remove indicates whether the parameter should be stripped from the incoming request
-	remove bool
+	// removeHeader indicates whether the parameter should be stripped from the incoming request
+	removeHeader bool
```
Comment (deads2k) [inline pkg/auth/authenticator/request/paramtoken/paramtoken.go]: This isn't removing a header.  removeToken? removeParameter?

**Comment by deads2k** [comment]
The change looks fine, but why would we bother to do this?

**Comment by liggitt** [comment]
To make sure our auth layer is the only one turning an auth token/header/param into a user. Not doing this caused problems when proxying to kubernetes and having original auth headers make it through to the backend

**Comment by liggitt** [comment]
[merge]

**Pull request merged**